core/src/main/kotlin/xtdb/arrow/DenseUnionVector.kt

```diff
@@ -138,6 +138,20 @@ class DenseUnionVector(
         valueCount++
     }
 
+    override fun writeNull() {
+        var legWriter : LegWriter? = null
+        for (i in legVectors.indices) {
+            val leg = legVectors[i]
+            if (leg.nullable) {
+                legWriter = LegWriter(i.toByte(), leg)
+                break
+            }
+        }
+        legWriter?.writeNull() ?: run {
+            TODO("auto-creation: null vs ${legVectors.map { it.name }}")
+        }
+    }
+
```

Comment:
The controversial bit. If we start with `writeNull`, we should likely also implement the others.